File: 2022-2023/sourcecode/lab/lab5/ViewPagerRecyclerViewAdapterJava/app/src/main/java/pl/edu/uwr/pum/viewpagerrecyclerviewadapterjava/PagerAdapter.java
package pl.edu.uwr.pum.viewpagerrecyclerviewadapterjava;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class PagerAdapter extends RecyclerView.Adapter<PagerAdapter.ViewHolder> {
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(
                        R.layout.item_view, parent, false
                )
        );
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(MainActivity.values[position]);
    }

    @Override
    public int getItemCount() {
        return MainActivity.values.length;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView textView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            textView = itemView.findViewById(R.id.textView);
        }

        public void bind(String text){
            textView.setText(text);
        }
    }
}

File: 2022-2023/sourcecode/lab/lab5/ViewPagerRecyclerViewAdapterKotlin/app/src/main/java/pl/edu/uwr/pum/viewpagerrecyclerviewadapterkotlin/PagerAdapter.kt
package pl.edu.uwr.pum.viewpagerrecyclerviewadapterkotlin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class PagerAdapter : RecyclerView.Adapter<PagerAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        val textView: TextView = view.findViewById(R.id.textView)

        fun bind(text: String){
            textView.text = text
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context).inflate(
                R.layout.item_view, parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(MainActivity.values[position])
    }

    override fun getItemCount(): Int = MainActivity.values.size
}